Content.Server/Ninja/Systems/SpaceNinjaSystem.cs

```diff
@@ -0,0 +1,358 @@
+    }
+
+    /// <summary>
+    /// Completes the objective, makes announcement and adds rule of a random threat.
+    /// </summary>
+    public void CallInThreat(EntityUid uid)
+    {
+        var config = RuleConfig();
+        if (config.Threats.Count == 0 || !GetNinjaRole(uid, out var role) || role.CalledInThreat)
+            return;
+
+        role.CalledInThreat = true;
+
+        var threat = _random.Pick(config.Threats);
+        _gameTicker.StartGameRule(threat.Rule, out _);
+        _chat.DispatchGlobalAnnouncement(Loc.GetString(threat.Announcement), playSound: false, colorOverride: Color.Red);
+    }
+
+    /// <summary>
+    /// Set up ninja when created.
+    /// </summary>
+    private void OnNinjaStartup(EntityUid uid, SpaceNinjaComponent comp, ComponentStartup args)
+    {
+        // start with internals on, only when spawned by event. antag control ninja won't do this due to component add order.
+        _internals.ToggleInternals(uid, uid, true);
+
+        // inject starting implants if made ninja in antag ctrl
+        AddImplants(uid);
+    }
+
+    /// <summary>
+    /// Add configured implants to the ninja.
+    /// </summary>
+    /// <remarks>
+    /// Could be replaced with job specials ImplantSpecial if ninja became a job somehow.
+    /// </remarks>
+    private void AddImplants(EntityUid uid)
+    {
+        var config = RuleConfig();
+        var coords = Transform(uid).Coordinates;
+        foreach (var id in config.Implants)
+        {
+            var implant = Spawn(id, coords);
+
+            if (!TryComp<SubdermalImplantComponent>(implant, out var implantComp))
+                return;
+
+            _implants.ForceImplant(uid, implant, implantComp);
+        }
+    }
+
+    /// <summary>
+    /// Greets the ninja when a ghost takes over a ninja, if that happens.
+    /// </summary>
+    private void OnNinjaMindAdded(EntityUid uid, SpaceNinjaComponent comp, MindAddedMessage args)
+    {
+        if (TryComp<MindContainerComponent>(uid, out var mind) && mind.Mind != null)
+            GreetNinja(mind.Mind);
+    }
+
+    /// <summary>
+    /// Set up everything for ninja to work and send the greeting message/sound.
+    /// </summary>
+    /// <remarks>
+    /// Currently this adds the ninja to traitors, this should be removed when objectives are separated from traitors.
+    /// </remarks>
+    private void GreetNinja(Mind.Mind mind)
+    {
+        if (!_mind.TryGetSession(mind, out var session) || mind.OwnedEntity == null)
+            return;
+
+        // make sure to enable the traitor rule for the sweet greentext
+        var traitorRule = EntityQuery<TraitorRuleComponent>().FirstOrDefault();
+        if (traitorRule == null)
+        {
+            // TODO: fuck me this shit is awful, see TraitorRuleSystem
+            _gameTicker.StartGameRule("Traitor", out var ruleEntity);
+            traitorRule = Comp<TraitorRuleComponent>(ruleEntity);
+        }
+
```

Comment:
Not a fan of ninja being tied to traitor, just inherit it or some shit if you really want it idk but as above gamerules should be able to added infinitely.

Reply:
wouldnt the only way to have the round summary be copy paste like 300 lines from traitor system, on top of all the objective logic and stuff